### PR TITLE
Implement MlConfigIndexMappingsFullClusterRestartIT test which verifies that .ml-config index mappings are properly updated during cluster upgrade

### PR DESCRIPTION
**Pull request by przemekwitek**
This PR adds a test which verifies that .ml-config index mappings are properly updated during cluster upgrade from version 7 to 8. 
It verifies the existence of data frame analytics mappings which were introduced in 7.3.
Data frame analytics mappings may show up either in:
- old cluster if old cluster version is on or after 7.3
or
- new cluster if old cluster version is before 7.3. This is the branch that really verifies the mappings update logic.

Related to #44263

### CODE REVIEW

**Comment by przemekwitek** [comment]
run elasticsearch-ci/2
run elasticsearch-ci/bwc

**Comment by przemekwitek** [comment]
run elasticsearch-ci/bwc

**Review by benwtrent — APPROVED** [review]
On your backport, you may have to do some version restrictions (or add handlers for versions before 7.2.0).

**Review by davidkyle — APPROVED** [review]
LGTM

File: x-pack/qa/full-cluster-restart/src/test/java/org/elasticsearch/xpack/restart/MlConfigIndexMappingsFullClusterRestartIT.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+            // trigger .ml-config index mappings update
+            createAnomalyDetectorJob(NEW_CLUSTER_JOB_ID);
+            // assert that the mappings are updated
+            assertThat(mappingsForDataFrameAnalysis(), is(equalTo(EXPECTED_DATA_FRAME_ANALYSIS_MAPPINGS)));
+        }
+    }
+
+    private void assertThatMlConfigIndexDoesNotExist() throws IOException {
+        Request getIndexRequest = new Request("GET", ".ml-config");
+        try {
```
[the remark this answers is not in the document]
Comment (przemekwitek) [reply x-pack/qa/full-cluster-restart/src/test/java/org/elasticsearch/xpack/restart/MlConfigIndexMappingsFullClusterRestartIT.java]: Done.

**Pull request merged**